### ISSUE TITLE
Code is not generated after upgrading dependencies

### ISSUE THREAD

**Issue by stefan-s29** [simolus3/moor#2079]
I just upgraded all of my project dependencies, including an update of Drift from `1.7.1` to `2.1.0`. My generated Drift file now had an issue, so I deleted it and wanted to regenerate it, but when I run `flutter pub run build_runner build`, nothing happens. All I get is this output:

```
[INFO] Generating build script...
[INFO] Generating build script completed, took 320ms

[INFO] Initializing inputs
[INFO] Reading cached asset graph...
[INFO] Reading cached asset graph completed, took 90ms

[INFO] Checking for updates since last build...
[INFO] Checking for updates since last build completed, took 751ms

[INFO] Running build...
[INFO] Running build completed, took 20ms

[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 56ms

[INFO] Succeeded after 87ms with 0 outputs (0 actions)
```

I am not sure if this has to do with other dependencies. Might as well be stupidity on my part, but I re-read the "Getting started" page to make sure that I am not missing something... I do have to run the command in my main project folder, right?

My database file is located in `lib/model/database/myapp_db.dart` and looks like this:

```
import 'dart:io';

import 'package:drift/drift.dart';
import 'package:drift/native.dart';
import 'package:path/path.dart' as path;
import 'package:path_provider/path_provider.dart';

part './myapp_db.g.dart';

@DriftDatabase(include: {'queries.drift'})
class MyAppDB extends _$MyAppDB {
  MyAppDB() : super(_openConnection());

  @override
  int get schemaVersion => 1;
  ```
  
Here is my list of dependencies:

```
environment:
  sdk: ">=2.7.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter

  bloc: ^8.1.0
  cupertino_icons: ^1.0.2
  drift: ^2.1.0
  equatable: ^2.0.5
  flutter_bloc: ^8.1.1
  flutter_webview_plugin:
    git: https://github.com/nuc134r/flutter_webview_plugin.git
  http: ^0.13.4
  meta: ^1.3.0
  multiple_stream_builder: ^3.0.2
  path: ^1.8.0
  path_provider: ^2.0.4
  provider: ^6.0.2
  rxdart: ^0.27.2
  sqlite3_flutter_libs: ^0.5.10
  flow_builder: ^0.0.6

dev_dependencies:
  drift_dev: ^2.1.0
  build_runner: ^2.1.4
  flutter_test:
    sdk: flutter
```

Any help is appreciated!

**Comment by simolus3** [comment]
Do you have a `build.yaml` file? If so, could you post it? Does `flutter pub run build_runner clean` help (it probably doesn't, but let's try :D)? After you've cleaned the build state, could you run the build with `--verbose` and see if anything unexpected comes up?

After running the build, does `find .dart_tool/build/generated/ | grep '\.part$'` find anything? Perhaps `source_gen` doesn't like the relative import for the part statement anymore, does `part 'myapp_db.g.dart';` work?

**Comment by stefan-s29** [comment]
No, I don't have a `build.yaml`. However, the `build_runner clean` command did help! The `myapp_db.g.dart` file was not generated right away, at first I got a few warnings (which I partly fixed) and there was only a `myapp_db.g.drift.dart` file, but when I had already started to type this post I somehow figured it out...

Unfortunately I don't remember what exactly was the crucial step. I only remember invalidating the cache and restarting Android Studio, but I am pretty sure I had to try a few more times after that... Anyway, it works now! Thanks for your help!

Sorry that I cannot provide more useful information, because there seems to be some issue either with Android Studio, Flutter or Drift...
